core/src/com/unciv/logic/automation/unit/SpecificUnitAutomation.kt

```diff
@@ -97,29 +97,69 @@ object SpecificUnitAutomation {
     }
 
     fun automateSettlerActions(unit: MapUnit, tilesWhereWeWillBeCaptured: Set<Tile>) {
-        if (unit.civ.gameInfo.turns == 0) {   // Special case, we want AI to settle in place on turn 1.
+        // If we don't have any cities, we are probably at the start of the game with only one settler
+        // If we are at the start of the game lets spend a maximum of 3 turns to settle our first city
+        // As our turns progress lets shrink the area that we look at to make sure that we stay on target
+        // If we have gone more than 3 turns without founding a city lets search a wider area
+        // TODO: Figure out a way to not use turns as it might not be the best metric, what if we are a civ starting in the middle of a game?
+        val rangeToSearch = if (unit.civ.cities.isEmpty() && unit.civ.gameInfo.turns < 4) (3 - unit.civ.gameInfo.turns).coerceAtLeast(1) else null
+        
+        // It's possible that we'll see a tile "over the sea" that's better than the tiles close by, but that's not a reason to abandon the close tiles!
+        // Also this lead to some routing problems, see https://github.com/yairm210/Unciv/issues/3653
+        val bestTilesToFoundCityReturn = CityLocationTileRanker.getBestTilesToFoundCity(unit, rangeToSearch)
+        val bestTilesToFoundCity = bestTilesToFoundCityReturn.first
+        val bestTile = bestTilesToFoundCityReturn.second.first
+        val bestTileRank = bestTilesToFoundCityReturn.second.second
```

Comment:
This is what I meant about a new data class. Just putting this in a class with named fields would make this instantly less esoteric

Reply:
I mean we won't even need these as separate vals, we can just reference A.B directly and it will be readable

Reply:
The only reason I was doing returning the best tile separately was for performance, I guess. If we compute the best value while ranking the tiles it should take much less time than taking maxOf() or something like that. 
But yes, I could store the best tile and get the best tile rank that way. Would a Pair of the map and best tile still be too much clutter?

Reply:
I think that a one-line data class is just so easy to make that there's no reason to use pairs at all :)